src/pretix/plugins/sendmail/models.py

```diff
@@ -278,16 +279,27 @@ def save(self, **kwargs):
 
         create_sms = []
         if self.event.has_subevents:
-            for se in self.event.subevents.annotate(has_sm=Exists(ScheduledMail.objects.filter(
-                    subevent=OuterRef('pk'), rule=self))).filter(has_sm=False):
-                sm = ScheduledMail(rule=self, subevent=se, event=self.event)
-                sm.recompute()
-                create_sms.append(sm)
+            if self.subevent:
+                ScheduledMail.objects.get_or_create(rule=self, subevent=self.subevent, event=self.event)
+            else:
+                for se in self.event.subevents.annotate(has_sm=Exists(ScheduledMail.objects.filter(
+                        subevent=OuterRef('pk'), rule=self))).filter(has_sm=False):
+                    sm = ScheduledMail(rule=self, subevent=se, event=self.event)
+                    sm.recompute()
+                    create_sms.append(sm)
             ScheduledMail.objects.bulk_create(create_sms)
         else:
             ScheduledMail.objects.get_or_create(rule=self, event=self.event)
 
         if not is_creation:
+            if self.subevent:
+                keep_states = [ScheduledMail.STATE_COMPLETED,
+                               ScheduledMail.STATE_FAILED]  # we keep rules that have already been sent at least partially
+                ScheduledMail.objects.filter(rule=self).exclude(
+                    subevent=self.subevent,
+                    state__in=keep_states
```

Comment:
Sorry, I did not catch this earlier, but I think this needs to be subevent=self.subevent OR state__in=keep_states. Should be doable with `Q`-objects. :)